core/include/drivers/firewall.h

```diff
@@ -0,0 +1,113 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2024, STMicroelectronics
+ */
+
+#ifndef __DRIVERS_FIREWALL_H
+#define __DRIVERS_FIREWALL_H
+
+#include <compiler.h>
+#include <drivers/firewall_device.h>
+#include <mm/core_memprot.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <tee_api_defines.h>
+#include <types_ext.h>
+
+struct firewall_controller_ops;
+
+/**
+ * struct firewall_controller - Information on firewall controller supplying
+ * services
+ *
+ * @ops: Operation handlers
+ * @name: Name of the firewall controller
+ * @base: Base address of the firewall controller
+ * @priv: Private data of the firewall controller
+ */
+struct firewall_controller {
+	const struct firewall_controller_ops *ops;
+	const char *name;
+	struct io_pa_va *base;
+	void *priv;
+};
+
+/**
+ * struct firewall_controller_ops - Firewall controller operation handlers
+ *
+ * @set_conf: Set given firewall configuration
```

Comment:
nitpicking:for consistency, either add `Callback used to ` here, or (my suggestion) remove those word from the lines below.